libdnf/dnf-context.cpp

```diff
@@ -1409,6 +1426,22 @@ dnf_context_set_install_weak_deps(gboolean enabled)
     mainConf.install_weak_deps().set(libdnf::Option::Priority::RUNTIME, enabled);
 }
 
+/**
+ * dnf_context_set_allow_vendor_change:
+ * @context: a #DnfContext instance.
+ * @vendorchange: %TRUE to allow changing vendors
+ *
+ * Enables or disables automatic changing of vendors in a transaction.
+ *
+ * Since: 0.55.2
+ */
+void
+dnf_context_set_allow_vendor_change(DnfContext *context, gboolean vendorchange)
```

Comment:
It will work with microdnf, but it is redundant. See https://github.com/rpm-software-management/microdnf/pull/88/files#r526049471.
It will not work with Packagekit and may be even for Microdnf (sack is not available).
I suggest that you can return to the original code where you modified main conf. Then in `dnf_context_setup_sack_with_flags` you can set vendor change value according to value in main conf.
What do you think?
PS: Any time when you use sack from context you have to consider that it could be NULL. Specially with packagekit this is often the right suggestion, because it handles sacks, goals outside of the context.

Reply:
That makes sense to me. Let me roll back to that and try that.